include/ddsrouter/participant/implementations/rtps/impl/DiscoveryServerParticipant.ipp

```diff
@@ -353,32 +337,93 @@ void DiscoveryServerParticipant<ConfigurationType>::enable_tls(
     descriptor->tls_config.add_option(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSOptions::NO_SSLV2); // not safe
 
-    // CA certificate
-    descriptor->tls_config.verify_file = get_param(tls_config, TLS_CA_TAG);
     // Perform verification of the server
     descriptor->tls_config.add_verify_mode(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_PEER);
 
-    if (client_only)
+    if (client)
     {
-        // Fail verification if the server has no certificate
-        descriptor->tls_config.add_verify_mode(
-            eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_FAIL_IF_NO_PEER_CERT);
+        if (!tls_configuration->can_be_client())
+        {
+            logError(DDSROUTER_DISCOVERYSERVER_PARTICIPANT,
+                    "TLS Configuration expected a Client configuration.");
+            throw ConfigurationException("TLS Configuration expected a Client configuration.");
+        }
+        else
+        {
+            enable_tls_client(descriptor, tls_configuration, true);
+        }
     }
     else
     {
-        // Password
-        descriptor->tls_config.password = get_param(tls_config, TLS_PASSWORD_TAG, false);
-        // Private key
-        descriptor->tls_config.private_key_file = get_param(tls_config, TLS_PRIVATE_KEY_TAG);
-        // DDS-Router certificate
-        descriptor->tls_config.cert_chain_file = get_param(tls_config, TLS_CERT_TAG);
-        // DH
-        descriptor->tls_config.tmp_dh_file = get_param(tls_config, TLS_DHPARAMS_TAG);
+        if (!tls_configuration->can_be_server())
+        {
+            logError(DDSROUTER_DISCOVERYSERVER_PARTICIPANT,
+                    "TLS Configuration expected a Server configuration.");
+            throw ConfigurationException("TLS Configuration expected a Server configuration.");
+        }
+        else
+        {
+            // Add configuration for server
+            enable_tls_server(descriptor, tls_configuration);
+
+            // In case it could also be client, add tls config
+            if (tls_configuration->can_be_client())
```

Comment:
This should be a requirement, so please update the comment and throw an exception if can't be client.

Reply:
We decided is not